content/en/docs/tasks/network/extend-service-ip-ranges.md

```diff
@@ -196,3 +200,99 @@ kubectl get servicecidr newcidr1
 ```
 Error from server (NotFound): servicecidrs.networking.k8s.io "newcidr1" not found
 ```
+
+## Kubernetes Service CIDR Policies
+
+Cluster administrators can implement policies to control the creation and
+modification of ServiceCIDR resources within the cluster. This allows for
+centralized management of the IP address ranges used for Services and helps
+prevent unintended or conflicting configurations. Kubernetes provides mechanisms
+like Validating Admission Policies to enforce these rules.
+
+### Preventing Unauthorized ServiceCIDR Creation/Update using Validating Admission Policy
+
+There can be situations that the cluster administrators want to restrict the
+ranges that can be allowed or to completely deny any changes to the cluster
+Service IP ranges.
+
+{{< note >}}
+The default "kubernetes" ServiceCIDR is created by the kube-apiserver
+to provide consistency in the cluster and is required for the cluster to work,
+so it always must be allowed.
+{{</ note >}}
+
+#### Restrict Service CIDR ranges to some specific ranges
+
+The following example demonstrates how to use a `ValidatingAdmissionPolicy` and
+its binding to restrict the Service CIDR ranges that can be created or updated.
+It uses the [CEL expression](https://kubernetes.io/docs/reference/using-api/cel/) `object.spec.cidrs.all(i, variables.allowed.exists(j, cidr(j).containsCIDR(i)))`
+to check that all of the CIDR values specified in the ServiceCIDR obejct are
+subranges of one of the two defined by the variable `allowed`. So it would allow
+a ServiceCIDR with `cidrs: ['10.96.1.0/24']` or `cidrs:
+['2001:db8:0:0:ffff::/80', '10.96.0.0/20']` but would not allow a ServiceCIDR
+with `cidrs: ['172.20.0.0/16']`.
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "servicecidrs.default"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["networking.k8s.io"]
+      apiVersions: ["v1","v1beta1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["servicecidrs"]
+  matchConditions:
+  - name: 'exclude-default-servicecidr'
+    expression: "object.metadata.name != 'kubernetes'"
+  variables:
+  - name: allowed
+    expression: "['10.96.0.0/16','2001:db8::/64']"
+  validations:
+  - expression: "object.spec.cidrs.all(i , variables.allowed.exists(j , cidr(j).containsCIDR(i)))"
```

Comment:
still needs better variable names. `i` and `j` are indexes, not values.
It's annoying that we can't use `cidr` for the first variable, but even `c` and `a` would be better than `i` and `j`.

Reply:
changed to `"object.spec.cidrs.all(currentCIDR, variables.allowed.exists(allowedCIDR, cidr(allowedCIDR).containsCIDR(currentCIDR)))"`